tests/router.test.ts

```diff
@@ -135,6 +135,37 @@ describe("Router lookup", function () {
       "route/with/trailing/slash": { path: "route/with/trailing/slash/" },
     });
   });
+
+  describe("routes with lower maxDepth should be considered too", function () {
+    testRouter(
+      [
+        "/",
+        "/:packageAndRefOrSha",
```

Comment:
This one triggers the `if (node && node.placeholderChildren.length > 1) {` condition.